docs/mismatches.md

```diff
@@ -0,0 +1,255 @@
+        0xc0000094 encountered at address 0x19c1f3b0105. The process has been
+        returned to the state before expression evaluation.` This process may
+        take more than 10 seconds!
+
+*   Invalid memory access
+
+    ```cpp
+    > *invalid_ptr
+    > invalid_ptr->field
+    > &(invalid_ptr)->ref_field
+    > &(invalid_ptr)->virtual_field
+    ```
+
+    *   The most frequently reported mismatch by fuzzer.
+    *   In most cases LLDB outputs `supposed to interpret, but failed:
+        Interpreter couldn't read from memory`. It could also results with
+        "process being returned to the state before evaluation" which can take
+        several seconds.
+    *   The first two expressions were not reproducible in fuzzer since
+        [#63](https://github.com/google/lldb-eval/pull/63).
+
+*   Casting negative floats to unsigned integers
+
+    ```cpp
+    > (unsigned int)-1.5
+
+    // lldb      : 0
+    // lldb-eval : 4294967295
+
+    > __log2(-1.5)  // takes unsigned int
+
+    // lldb      : <random-value>  (computes __log2 of a random unsigned int)
+    // lldb-eval : 31              (computes __log2 of 4294967295)
+    ```
+
+    *   Frequently reported by fuzzer
+
+*   Casting to different pointer type and dereferencing
+
+    ```cpp
+    int x = 2;
+
+    > (int) *(bool*) &x
+
+    // lldb      : 2
+    // lldb-eval : 1
+    ```
+
+    *   Encountered: 2021 Feb (reported by fuzzer)
+
+## Possible bugs in LLDB
+
+*   ```cpp
+    // Test is class or struct
+
+    > sizeof(Test)
+    // lldb      : 'Test' has unknown type; cast it to its declared type
+    // lldb-eval : 24
+    ```
+
+    *   Note: This was reproducible only on Windows, when `Test` was `struct` or
+        `class` (e.g. there's no problem with `union`) that was defined globally
+        under no-namespace and contained at least one non-static member.
+    *   Workaround is to explicitly use `struct` or `class` keyword, e.g.
+        `sizeof(struct Test)`.
+    *   Encountered: 2021 Mar
+
+    Similar example is:
+
+    ```cpp
+    > (Test*)ptr
+    // lldb      : expression failed to parse, fixed expression suggested:
+    //             (struct TestStruct*)0
+    ```
+
+*   `this` isn't compatible with its own type
+
+    ```cpp
+    // Test binary:
+    struct TestStruct { ... };
+    TestStruct ts;
+
+    // Using APIs:
+    lldb::SBValue ts1 = frame.FindVariable("ts");
+    lldb::SBValue ts2 = frame.EvaluateExpression("ts");
+
+    > lldb_eval::EvaluateExpression(ts1, "this == (TestStruct*)this", ...);
+    // OK (true)
+
+    > lldb_eval::EvaluateExpression(ts2, "this == (TestStruct*)this", ...);
+    // comparison of distinct pointer types ('TestStruct *' and 'TestStruct *')
+    ```
+
+    *   TODO: Investigate the exact reason for this bug. This could be related
+        to equality operator of `lldb::SBType` not working as expected.
+    *   Encountered: 2021 Apr 14 (reported by fuzzer while supporting evaluation
```

Comment:
Let's have consistent ordering of items. Right now "Bugs in lldb-eval" seems to be ordered from newest to oldest, but "Possible bugs in LLDB" is the other way around. I prefer from oldest to newest (i.e. oldest at the top, adding new items to the bottom)

Reply:
Reordered from oldest to newest.